BootloaderCommonPkg/Library/TpmLib/TpmLib.c

```diff
@@ -161,23 +219,26 @@ TpmPcrBankCheck (
   UINT32                ActivePcrBanks;
 
   Status = Tpm2GetCapabilitySupportedAndActivePcrs (&TpmHashAlgorithmBitmap, &ActivePcrBanks);
+
   if (Status == EFI_SUCCESS) {
     // Update Active PCR banks info in TPM Library data
     TpmLibSetActivePcrBanks (ActivePcrBanks);
 
     // Check if TPM supports SHA256 PCR bank
-    if ((TpmHashAlgorithmBitmap & HASH_ALG_SHA256) == 0) {
-      DEBUG ((DEBUG_ERROR, "SHA256 hash algorithm is NOT supported. This should never happen with TPM2.0 device. \n"));
+    if (!(TpmHashAlgorithmBitmap & HASH_ALG_SHA256) &&  !(TpmHashAlgorithmBitmap & HASH_ALG_SHA384) && !(TpmHashAlgorithmBitmap & HASH_ALG_SM3_256)){
+      DEBUG ((DEBUG_ERROR, "Hash algorithm is NOT supported. This should never happen with TPM2.0 device. \n"));
       return EFI_UNSUPPORTED;
     }
 
     // Check if 'only' SHA256 PCR bank is enabled
-    if (ActivePcrBanks == HASH_ALG_SHA256) {
-      DEBUG ((DEBUG_INFO, "SHA256 PCR Bank is enabled. \n"));
+    if (ActivePcrBanks == GetTcgHashEnabled()) {
+      DEBUG ((DEBUG_INFO, "Bootloader requested PCR Bank is enabled. \n"));
     } else {
-      Status = Tpm2PcrAllocateBanks (NULL, TpmHashAlgorithmBitmap, HASH_ALG_SHA256);
+      Status = Tpm2PcrAllocateBanks (NULL, TpmHashAlgorithmBitmap, GetTcgHashEnabled());
       if (Status == EFI_SUCCESS) {
-        DEBUG ((DEBUG_INFO, "Successfully enabled SHA256 PCR Bank. It will be available after next platform reset. \n"));
+        DEBUG ((DEBUG_INFO, "Successfully enabled requested PCR Bank. It will be available after next platform reset. \n"));
+
+         ResetSystem(EfiResetCold);
```

Comment:
I think this was discussed earlier and initiating a 'reset' was NOT recommended. So, let's skip it.